packages/core-data/src/actions.js

```diff
@@ -115,14 +115,97 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Returns an action object that triggers an
+ * edit to an entity record.
+ *
+ * @param {string} kind           Kind of the edited entity record.
+ * @param {string} name           Name of the edited entity record.
+ * @param {number} recordId       Record ID of the edited entity record.
+ * @param {Object} edits          The edits.
+ *
+ * @return {Object} Action object.
+ */
+export function* editEntityRecord( kind, name, recordId, edits ) {
+	const { transientEdits = {} } = yield select( 'getEntity', kind, name );
+	const record = yield select( 'getEntityRecord', kind, name, recordId );
+	const editedRecord = yield select(
+		'getEditedEntityRecord',
+		kind,
+		name,
+		recordId
+	);
+
+	const edit = {
+		kind,
+		name,
+		recordId,
+		// Clear edits when they are equal to their persisted counterparts
+		// so that the property is not considered dirty.
+		edits: Object.keys( edits ).reduce( ( acc, key ) => {
```

Comment:
Isn't this behavior already done in the reducer? at least it feels like a better place for it.

Reply:
I needed access to the entity, I guess I can also use a higher order reducer for this.